src/main/scala/mesosphere/marathon/tasks/ResourceUtil.scala

```diff
@@ -109,10 +114,10 @@ object ResourceUtil {
     * Deduct usedResources from resources by matching them by name.
     */
   def consumeResources(resources: Iterable[Resource], usedResources: Iterable[Resource]): Iterable[Resource] = {
-    val usedResourceMap = usedResources.map(resource => resource.getName -> resource).toMap
+    val usedResourceMap = usedResources.map(resource => ResourceId(resource) -> resource).toMap
 
     resources.flatMap { resource =>
-      usedResourceMap.get(resource.getName) match {
+      usedResourceMap.get(ResourceId(resource)) match {
         case Some(usedResource) =>
           if (resource.getType != usedResource.getType) {
```

Comment:
Would it make sense to also add the resource type into the ResourceId, so that we don't have to handle that case specifically?